code/computational_geometry/src/DDALineDrawingAlgorithm.cpp

```diff
@@ -0,0 +1,61 @@
+#include <iostream>
+#include <math.h>
+
+class DDALineDrawingAlgorithm
+{
+public:
+    void ddaLineDrawing();
+
+    void getCoordinates();
+
+private:
+    int x1_, x2_, y1_, y2_;
+
+};
+
+void DDALineDrawingAlgorithm::ddaLineDrawing()
+{
+    //calculating range for line between start and end point
+    int dx = x2_ - x1_;
+    int dy = y2_ - y1_;
+    // calculate steps required for creating pixels
+    int step = abs(abs(dx) > abs(dy) ? dx : dy);
+    // calculate increment in x & y for each steps
+    float xIncrement = (dx) / (float)step;
+    float yIncrement = (dy) / (float)step;
+    // drawing  pixel for each step
+    float x = x1_;
+    float y = y1_;
+    putpixel((x), (y),GREEN);   //this putpixel is for very first pixel of the line
```

Comment:
Even if you don't include the library, use a more modern replacement. Also, replace `math.h` with `cmath`.

Reply:
As this is the basic algorithm hence it should be implemented by using a basic function which would help the early bird learner to grasp it quickly and I will change the library for inbuilt math function.